docs/kb/semgrep-ci/jenkins-diff-scans.md

```diff
@@ -0,0 +1,185 @@
+              git checkout $GIT_BRANCH
+           '''
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -e SEMGREP_BASELINE_REF=$(git merge-base $GIT_BRANCH $CHANGE_TARGET) \
+            -e SEMGREP_PR_ID = "${env.CHANGE_ID}"
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+    stage('semgrep-scan') {
+      when {
+        branch "main"
+      }
+      steps {
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+  post {
+    // Clean after build
+    always {
+      cleanWs()
+    }
+  }
+}
+```
+
+The Jenkinsfile defines two Semgrep stages, each of which is run for certain branches: a diff-aware scan for PR branches, and a full scan for the main branch. This diff-aware scan configuration uses a computed merge base, rather than setting the merge base to the default branch. This is more similar to how Semgrep performs in GitHub actions. Setting the SEMGREP_REPO_NAME and SEMGREP_PR_ID allows Semgrep to identify the connected project and related PR.
+
+In order to compute the merge-base, the pipeline performs some additional git commands to make the default branch available to git for computation, and cleans the workspace afterwards so those commands can always run successfully in a clean workspace.
+
+:::info
+Using a computed merge base is strongly recommended. If you instead set `SEMGREP_BASELINE_REF` to `main` or `master`, you may see spurious findings in diff-aware scans if the remote branch has been updated independently of the PR branch, or the branch may not be available locally unless you perform a `git fetch` or `git checkout` as shown in this example.
+:::
+
+### Configure the Multibranch pipeline
+
+- Branch Sources: GitHub
+    - Credentials: (select the appropriate credentials for your project)
+    - Repository HTTPS URL: `https://github.com/<namespace>/<project>`
+- Behaviors:
+    - Discover branches
+        - Strategy: exclude branches that are also filed as PRs
+    - Discover pull requests from origin
+        - The current pull request revision
+    - Check out to matching local branch
+    - Property strategy: all branches get the same properties
+- Build Configuration: by Jenkinsfile
+- Scan Repository Triggers: periodically if not otherwise run, 30 min (interval configurable)
+- Optional: Discard old items (7 days/7 items), so that you don’t lose logs for deleted branches immediately
+
+If you haven’t already, follow the [On GitHub](#on-github) instructions to configure a webhook to notify Jenkins about PR events.
```

Comment:
```suggestion
Finally, follow the [On GitHub](#on-github) instructions to configure a webhook to notify Jenkins about PR events.
```

Reply:
They don't need to do this if they already did - the instructions are in the first section.